simul/platform/aws.go

```diff
@@ -0,0 +1,282 @@
+
+	//*** Configure Master
+	fmt.Println("[+] Configuring Master")
+	for idx := 0; idx < len(configure); idx++ {
+		fmt.Println("       Exec:", idx, configure[idx])
+		_, err := master.Run(configure[idx])
+		if err != nil {
+			return err
+		}
+	}
+
+	//*** Configure Slaves
+	fmt.Println(*masterInstance.PublicIP)
+	slaveCmds := a.slaveCMDS.Configure(*masterInstance.PublicIP)
+	fmt.Println(*masterInstance.PublicIP)
+
+	fmt.Println("")
+	fmt.Println("")
+	fmt.Println("[+] Configuring Slaves:")
+
+	//addresses, syncs := aws.GenRemoteAddresses(slaveInstances)
+	var wg sync.WaitGroup
+
+	for _, slave := range slaveInstances {
+		//	node := lib.GenerateNode(cons, i, addr)
+		//	nodeAndSync := aws.NodeAndSync{node, syncs[i]}
+		wg.Add(1)
+		// TODO This might become a problem for large number of slaves,
+		// limit numebr of go-routines running concurrently if this is the case
+
+		go func(slave aws.Instance) {
+
+			slaveNodeController, err := aws.NewSSHNodeController(*slave.PublicIP, a.pemBytes, a.awsConfig.SSHUser)
+			if err != nil {
+				panic(err)
+			}
+			configureSlave(slaveNodeController, slaveCmds)
+			fmt.Println("    - Slave", *slave.PublicIP)
+			wg.Done()
+		}(*slave)
+		a.allSlaveNodes = append(a.allSlaveNodes, slave)
+	}
+	wg.Wait()
+	return nil
+}
+
+func configureSlave(slaveNodeController aws.NodeController, slaveCmds map[int]string) error {
+	slaveNodeController.Init()
+	defer slaveNodeController.Close()
+
+	for idx := 0; idx < len(slaveCmds); idx++ {
+		_, err := slaveNodeController.Run(slaveCmds[idx])
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (a *awsPlatform) Cleanup() error {
+	a.master.Close()
+	return a.aws.StopInstances()
+}
+
+func (a *awsPlatform) Start(idx int, r *lib.RunConfig) error {
+	nodePerInstances := r.Nodes
+	slaveNodes := a.allSlaveNodes[0:a.awsConfig.NbOfInstances]
+	aws.UpdateInstances(slaveNodes, nodePerInstances, a.cons)
+
+	writeRegFile(slaveNodes, a.masterCMDS.RegPath)
+
+	//*** Start Master
+	fmt.Println("[+] Registry file written to local storage(", r.Nodes, " nodes)")
+	fmt.Println("[+] Transfering registry file to Master")
+	a.master.CopyFiles(a.masterCMDS.RegPath)
+	shareRegistryFile := a.masterCMDS.ShareRegistryFile()
+	fmt.Println("[+] Master handel node:")
+	for i := 0; i < len(shareRegistryFile); i++ {
+		fmt.Println("       Exec:", i, shareRegistryFile[i])
+		_, err := a.master.Run(shareRegistryFile[i])
+		if err != nil {
+			return err
+		}
+	}
+
+	masterStart := a.masterCMDS.Start(a.masterAddr, a.awsConfig.NbOfInstances*nodePerInstances, a.awsConfig.MasterTimeOut)
+	fmt.Println("       Exec:", len(shareRegistryFile)+1, masterStart)
+	a.master.Start(masterStart)
+
+	//*** Starte slaves
+	var wg sync.WaitGroup
+	for _, n := range slaveNodes {
+		wg.Add(1)
+		go func(slaveNode aws.Instance) {
+			// TODO This might become a problem for large number of slaves,
+			// limit numebr of go-routines running concurrently if this is the case
+			a.startSlave(slaveNode, idx)
```

Comment:
Would it be possible to make the master node start all the slaves instances ? It would bring several advantages such as not overloading our laptop, use more capacities from the master slave (which I believe is under-used at the moment since it only waits), and probably faster as well since connections would be between amazon instances only - ie. in their network.

Reply:
Although it does not have to be implemented in this PR at all, it can be future work for this month if we see it's start getting slow on this path.

Reply:
Yeah this PR is already too big I will create separate issue

Reply:
https://github.com/ConsenSys/handel/issues/41